### PR TITLE
Prestashop detection

### PR DESCRIPTION
**Pull request by patrocle**
Prestashop detection without meta generator or "powered by"

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -5445,7 +5445,14 @@
 			"cats": [
 				6
 			],
-			"html": "Powered by <a\\s+[^>]+>PrestaShop",
+			"html":
+			[
+				"Powered by <a\\s+[^>]+>PrestaShop",
+				"freeProductTranslation\\;confidence:25",
```
Comment (gadcam) [inline src/apps.json]: A good practice is to ensure HTML patterns won't match plain text see https://github.com/AliasIO/Wappalyzer/wiki/Specification#too-generic-patterns
You should include an HTML tag in them.

**Comment by patrocle** [comment]
Hi,
Ok, prestashop have always 

``<script type="text/javascript">
var ajax_allowed = true;
var ajaxsearch = true;
var baseDir = 'http://www.xx.com/';
var baseUri = 'http://www.xx.com/';
var blocksearch_type = 'top';
var contentOnly = false;
var customizationIdMessage = 'Personnalisation';
var delete_txt = 'Supprimer';
var displayList = false;
var freeProductTranslation = 'Offert !';
var freeShippingTranslation = 'Livraison gratuite !';
var id_lang = 1;
var img_dir = 'http://www.xx.com/themes/cc/img/';
var instantsearch = false;
var isGuest = 0;
var isLogged = 0;
var loggin_required = 'Vous devez être connecté pour gérer votre liste de cadeaux';
var mywishlist_url = 'http://www.xx.com/module/blockwishlist/mywishlist';
var page_name = 'index';
var priceDisplayMethod = 1;
var priceDisplayPrecision = 2;
var quickView = true;
var removingLinkText = 'supprimer cet article du panier';
var roundMode = 2;
var search_url = 'http://www.xx.com/recherche';
var static_token = 'aaa';
var token = 'aaa';
var usingSecureMode = false;
var wishlistProductsIds = false;
</script>``

So I pull another request with 

``"<script>(?>(?!</?script>).)*(freeProductTranslation)(?>(?!</?script>).)*</script>",``
``"<script>(?>(?!</?script>).)*(freeProductTranslation)(?>(?!</?script>).)*</script>",``
``"<script>(?>(?!</?script>).)*(priceDisplayMethod)(?>(?!</?script>).)*</script>",``
``"<script>(?>(?!</?script>).)*(priceDisplayPrecision)(?>(?!</?script>).)*</script>"``

?

**Comment by louis-ako10** [comment]
Well that's better but it doesn't work for script with CDATA like 
`<script type="text/javascript">`
`/* <![CDATA[ */;`
` var priceDisplayMethod = 1;`
`/* ]]> */`
`</script>`

Only with 

`<script type="text/javascript">`
` var priceDisplayMethod = 1;`
`</script>`

**Comment by gadcam** [comment]
The "not" <code>(?!...)</code>  is not implemented in W. as far as I remember.
However you can use  env field see https://github.com/AliasIO/Wappalyzer/wiki/Specification#json-fields.
So it would look like
```js
"env": "^freeProductTranslation$"
```

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**